commands/azureCommands/acr-build-logs.ts

```diff
@@ -0,0 +1,381 @@
+        vscode.window.showErrorMessage('Cant open!');
+    });
+}
+/** Loads log text from remote url using azure blobservices */
+function openLog(url: string, title: string): void {
+    let blobInfo = getBlobInfo(url);
+    let blob: BlobService;
+    try {
+        blob = createBlobServiceWithSas(blobInfo.host, blobInfo.sasToken);
+    } catch (error) {
+        console.log(error);
+    }
+    try {
+        blob.getBlobToText(blobInfo.containerName, blobInfo.blobName, async (error, text, result, response) => {
+            if (response) {
+                createLogView(text, title);
+            } else {
+                console.log(error);
+            }
+        });
+    } catch (error) {
+        console.log(error);
+    }
+}
+/** Parses blob url into a readable form */
+function getBlobInfo(blobUrl: string): { accountName: string, endpointSuffix: string, containerName: string, blobName: string, sasToken: string, host: string } {
+    const items: string[] = blobUrl.slice(blobUrl.search('https://') + 'https://'.length).split('/');
+    const accountName: string = blobUrl.slice(blobUrl.search('https://') + 'https://'.length, blobUrl.search('.blob'));
+    const endpointSuffix: string = items[0].slice(items[0].search('.blob.') + '.blob.'.length);
+    const containerName: string = items[1];
+    const blobName: string = items[2] + '/' + items[3] + '/' + items[4].slice(0, items[4].search('[?]'));
+    const sasToken: string = items[4].slice(items[4].search('[?]') + 1);
+    const host: string = accountName + '.blob.' + endpointSuffix;
+    return { accountName, endpointSuffix, containerName, blobName, sasToken, host };
+}
+
+//# UTILS
+function makeBase64(str: string): string {
+    let buffer = new Buffer(str);
+    return buffer.toString('base64');
+}
+
+/** Obtains a function to filter logs to a single repository/image */
+function getFilterFunction(context: AzureRegistryNode | AzureRepositoryNode | AzureImageNode): (logEntry: Build) => boolean {
+    if (context instanceof AzureRegistryNode) {
+        return undefined;
+    } else if (context instanceof AzureRepositoryNode) {
+        return (logEntry: Build) => {
+            if (!logEntry.outputImages) {
+                return false;
+            } else if (logEntry.outputImages.length === 0) {
+                return false;
+            } else if (logEntry.outputImages.find((imgDescriptor) => {
+                if (!imgDescriptor) { return false; }
+                return imgDescriptor.repository === context.label;
+            })) {
+                return true
+            } else {
+                return false;
+            }
+        }
+    } else {
+        const tag: string = context.label.slice(context.label.search(':') + 1);
+        return (logEntry: Build) => {
+            if (!logEntry.outputImages) {
+                return false;
+            } else if (logEntry.outputImages.length === 0) {
+                return false;
```

Comment:
A lot of repeated code from the previous else if statement. Could potentially move to a new method